src/isar/services/utilities/scheduling_utilities.py

```diff
@@ -24,28 +30,192 @@ class SchedulingUtilities:
         self.logger = logging.getLogger("api")
 
     def get_state(self) -> States:
-        return self.queues.state.check()
+        """Return the current state of the state machine
+
+        Raises
+        ------
+        HTTPException 500 Internal Server Error
+            If the current state is not available on the queue
+        """
+        try:
+            return self.queues.state.check()
+        except Empty:
+            error_message: str = (
+                "Internal Server Error - Current state of the state machine is unknown"
+            )
+            self.logger.error(error_message)
+            raise HTTPException(
+                status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail=error_message
+            )
+
+    def get_mission(self, mission_id: int) -> Mission:
+        """Get the mission with mission_id from the current mission planner
+
+        Raises
+        ------
+        HTTPException 404 Not Found
+            If requested mission with mission_id is not found
+        HTTPException 500 Internal server error
+            If for some reason the mission can not be returned
+        """
+        try:
+            return self.mission_planner.get_mission(mission_id)
+        except HTTPError as e:
+            self.logger.error(e)
+            raise HTTPException(status_code=e.response.status_code)
+        except MissionNotFoundError as e:
+            self.logger.error(e)
+            raise HTTPException(
+                status_code=HTTPStatus.NOT_FOUND,
+                detail=f"Mission with id '{mission_id}' not found",
+            )
+        except MissionPlannerError as e:
+            self.logger.error(e)
+            raise HTTPException(
+                status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+                detail="Could not plan mission",
+            )
+
+    def assert_robot_is_capable_of_mission(
+        self, mission: Mission, robot_capabilities: List[str]
+    ):
+        """Assert that the robot is capable of performing this mission
+
+        Raises
+        ------
+        HTTPException 400 Bad request
+            If the robot is not capable of performing mission
+        """
+        is_capable: bool = True
+        missing_capabilities: Set[str] = set()
+        for task in mission.tasks:
+            for step in task.steps:
+                if not step.type in robot_capabilities:
+                    is_capable = False
+                    missing_capabilities.add(step.type)
+
+        if not is_capable:
+            error_message = (
+                f"Bad Request - Robot is not capable of performing mission."
+                f" Missing functionalities: {missing_capabilities}."
+            )
+            self.logger.error(error_message)
+            raise HTTPException(
+                status_code=HTTPStatus.BAD_REQUEST,
+                detail=error_message,
+            )
+
+        assert is_capable
+
+    def assert_state_machine_is_idle(self, state: States):
```

Comment:
Check if state machine is ready to receive a new mission rather than checking if it is idle

Reply:
Good idea, implemented